### PR TITLE
Initial templates

### PR DESCRIPTION
**Pull request by wtrocki**
## Progress

Posting for early feedback on the general layout. 
Some parameters etc can still change.
I have added minimal documentation. 
As templates require to be downstream I have created a separate repository. 
Using DataSync name in the Voyager-server project can be more than confusing for people.

The documentation explains my motivation. After chatting with @psturc we came with idea of having s2i and fully featured showcase template that works out of the box.

Any yaml fans here? 
Still testing/improving but waiting for feedback.

## Open Questions

- How to load image to the template - 
Do we have that added to OpenShift? Do we need to load that somehow magically to OpenShift UI?

- Is kube target valuable? Probably makes no sense?

- Do we still have mobile tab/category for those templates?

### CODE REVIEW

**Comment by wtrocki** [comment]
@wei-lee I remember we said that we kinda have the trend to support kubernetes deployment but in this case, I have mixed feelings about it because normally people will be like using helm to deploy they node app etc. Do you think there is value from this?

**Comment by wei-lee** [comment]
@wtrocki templates are different as itself only support on OpenShift. I don't think k8s supports it. So it is ok if it's OpenShift only, especially if it's using something like S2I.

> How to load image to the template -
Do we have that added to OpenShift? Do we need to load that somehow magically to OpenShift UI?

Just import it to the "openshift" namespace.

> Is kube target valuable? Probably makes no sense?

No, not for templates.

> Do we still have mobile tab/category for those templates?

It will be as the tab will show up once the mobileapp CRD is created.

File: kubernetes/deployment.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: extensions/v1beta1
```
Comment (wei-lee) [inline kubernetes/deployment.yml]: deployment is not an extension anymore, it is now part of the core k8s api. See https://kubernetes.io/docs/concepts/workloads/controllers/deployment/

**Review by wei-lee — APPROVED** [review]

File: kubernetes/deployment.yml (hunk 1 above)
Comment (wtrocki) [reply kubernetes/deployment.yml]: WOW. It did work for me, but going to change that

**Comment by wtrocki** [comment]
@wei-lee Thanks for responses. 
Going to remove kubernetes and add template to mobile category. 
Going to test both templates etc.

**Comment by wtrocki** [comment]
Screen Shot 2019-04-10 at 9 50 39 AM

Going to drop OpenShift and add `Mobile Developer Services DataSync`

**Comment by wtrocki** [comment]
Screen Shot 2019-04-10 at 12 41 04 PM

File: openshift/datasync-http.json [hunk 2]
```diff
@@ -0,0 +1,288 @@
+{
+    "kind": "Template",
+    "apiVersion": "v1",
+    "metadata": {
+      "name": "datasync-http",
+      "annotations": {
+        "openshift.io/display-name": "Mobile Developer Services DataSync",
```
Comment (darahayes) [inline openshift/datasync-http.json]:
```suggestion
        "openshift.io/display-name": "Mobile Developer Services Data Sync",
```

File: openshift/datasync-http.json [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+        "description": "Mobile Developer Services DataSync template",
```
Comment (darahayes) [inline openshift/datasync-http.json]:
```suggestion
        "description": "Mobile Developer Services Data Sync template",
```

File: openshift/datasync-http.json [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+        "tags": "sync, mobile, nodejs",
+        "iconClass": "icon-nodejs",
+        "openshift.io/long-description": "This template builds Node.js based applications integration with DataSync.",
```
Comment (darahayes) [inline openshift/datasync-http.json]:
```suggestion
        "openshift.io/long-description": "This template builds Node.js based applications integration with Data Sync.",
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+## DataSync Production Deployment
```
Comment (darahayes) [inline README.md:1]:
```suggestion
## Data Sync Production Deployment
```

File: README.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+Contains OpenShift templates for DataSync deployment
```
Comment (darahayes) [inline README.md:3]:
```suggestion
Contains OpenShift templates for Data Sync deployment
```

**Review by darahayes — COMMENTED** [review]
Looks good to me but I believe the name is 'Data Sync' and not 'DataSync.' Just a question: Do templates not support yaml format?

**Comment by wei-lee** [comment]
do developers actually need to change the namespace of the imagestream? Should it always be `opebshift`?

**Comment by wtrocki** [comment]
> Just a question: Do templates not support yaml format?

My mind do not support it. 
Do you think we should have yaml instead?

> do developers actually need to change the namespace of the imagestream? Should it always be opebshift?

I got that from s2i template so it seems that someone had use case  for that. 
I did not found anything so going to remove it.

**Review by darahayes — APPROVED** [review]

**Comment by wtrocki** [comment]
Linked PR: https://github.com/aerogear/ionic-showcase/pull/143/files

**Comment by wtrocki** [comment]
@darahayes @wei-lee I have moved templates to yml tested those on different environments and used recommended defaults.

**Pull request merged**